### PR TITLE
Escrow Ledger Node format

### PR DESCRIPTION
**Pull request by mDuo13**
Also contains some corrections to the transaction format page

### CODE REVIEW

File: content/reference-ledger-format.md [hunk 1]
```diff
@@ -239,6 +240,64 @@ The lower 64 bits of an Offer Directory's index represent the TakerPays amount d
 * The page number of this node. (Since 0 is the root DirectoryNode, this value is an integer 1 or higher.)
 
 
+## Escrow
+[[Source]<br>](https://github.com/ripple/rippled/blob/develop/src/ripple/protocol/impl/LedgerFormats.cpp#L90-L101 "Source")
```
Comment (wilsonianb) [inline content/reference-ledger-format.md]: I realize this source file isn't likely to change much, but I'd recommend linking to a specific commit.
https://github.com/ripple/rippled/blob/c6b6d82a754fe449cc533e18659df483c10a5c98/src/ripple/protocol/impl/LedgerFormats.cpp#L90-L101

File: content/reference-ledger-format.md [hunk 2]
```diff
@@ -239,6 +240,64 @@ The lower 64 bits of an Offer Directory's index represent the TakerPays amount d
...
+
+_(Requires the [Escrow Amendment](concept-amendments.html#paychan).)_
+
+The `Escrow` node type represents a held payment of XRP waiting to be executed or canceled. An [EscrowCreate transaction](reference-transaction-format.html#escrowcreate) creates an Escrow node in the ledger. A successful [EscrowFinish](reference-transaction-format.html#escrowfinish) or [EscrowCancel](reference-transaction-format.html#escrowcancel) transaction deletes the node.
+
+An Escrow node is associated with two addresses: the owner, who provides the XRP when creating the Escrow node; and the destination, where the XRP is paid when the held payment succeeds. If the held payment is canceled, the XRP returns to the owner. If the Escrow node has a [_crypto-condition_](https://tools.ietf.org/html/draft-thomas-crypto-conditions-02), the payment can only succeed if an EscrowFinish transaction provides the corresponding _fulfillment_ that satisfies the condition. (The only supported crypto-condition type is SHA-256-PREIMAGE.)
```
Comment (wilsonianb) [inline content/reference-ledger-format.md]: Maybe note that the owner and destination addresses can be the same.

Comment (wilsonianb) [reply content/reference-ledger-format.md]: Also consider linking `SHA-256-PREIMAGE` to https://tools.ietf.org/html/draft-thomas-crypto-conditions-02#section-8.1

File: content/reference-ledger-format.md [hunk 3]
```diff
@@ -239,6 +240,64 @@ The lower 64 bits of an Offer Directory's index represent the TakerPays amount d
...
+
+An Escrow node has the following fields:
+
+| Name              | JSON Type | [Internal Type][] | Description |
+|-------------------|-----------|---------------|-------------|
+| Account           | String | AccountID | The address of the owner (sender) of this held payment. This is the account that provided the XRP, and gets it back if the held payment is canceled. |
+| Destination       | String | AccountID | The destination address where the XRP is paid if the held payment is successful. |
+| Amount            | String | Amount    | The amount of XRP, in drops, to be delivered by the held payment. |
+| Condition         | String | VariableLength | _(Optional)_ A [PREIMAGE-SHA-256 crypto-condition](https://tools.ietf.org/html/draft-thomas-crypto-conditions-02#section-8.1), as hexadecimal. If present, the [EscrowFinish transaction][] must contain a fulfillment that satisfies this condition. |
+| CancelAfter       | Number | UInt32 | _(Optional)_ The time, in [seconds since the Ripple epoch](reference-rippled.html#specifying-time), after which this held payment can only be canceled. (Specifically, this is compared with the close time of the previous validated ledger.) |
```
Comment (wilsonianb) [inline content/reference-ledger-format.md]: Not super clear that the payment cannot be canceled before this time.

**Review by wilsonianb — APPROVED** [review]
LGTM :+1:

File: content/reference-ledger-format.md [hunk 4]
```diff
@@ -239,6 +240,64 @@ The lower 64 bits of an Offer Directory's index represent the TakerPays amount d
...
+
+_(Requires the [Escrow Amendment](concept-amendments.html#paychan).)_
+
+The `Escrow` node type represents a held payment of XRP waiting to be executed or canceled. An [EscrowCreate transaction](reference-transaction-format.html#escrowcreate) creates an Escrow node in the ledger. A successful [EscrowFinish](reference-transaction-format.html#escrowfinish) or [EscrowCancel](reference-transaction-format.html#escrowcancel) transaction deletes the node.
+
+An Escrow node is associated with two addresses: the owner, who provides the XRP when creating the Escrow node; and the destination, where the XRP is paid when the held payment succeeds. (The two addresses can be the same.) If the held payment is canceled, the XRP returns to the owner. If the Escrow node has a [_crypto-condition_](https://tools.ietf.org/html/draft-thomas-crypto-conditions-02), the payment can only succeed if an EscrowFinish transaction provides the corresponding _fulfillment_ that satisfies the condition. (The only supported crypto-condition type is [PREIMAGE-SHA-256](https://tools.ietf.org/html/draft-thomas-crypto-conditions-02#section-8.1).)
```
Comment (ryangyoung) [inline content/reference-ledger-format.md]: > An Escrow node is associated with two addresses: the owner, who provides the XRP when creating the Escrow node; and the destination, where the XRP is paid when the held payment succeeds.

Maybe break that into bullet points?

**Review by ryangyoung — APPROVED** [review]
Just had one very minor comment, otherwise LGTM.

File: content/reference-ledger-format.md [hunk 5]
```diff
@@ -239,6 +240,64 @@ The lower 64 bits of an Offer Directory's index represent the TakerPays amount d
 * The page number of this node. (Since 0 is the root DirectoryNode, this value is an integer 1 or higher.)
 
 
+## Escrow
+[[Source]<br>](https://github.com/ripple/rippled/blob/c6b6d82a754fe449cc533e18659df483c10a5c98/src/ripple/protocol/impl/LedgerFormats.cpp#L90-L101 "Source")
```
Comment (ryangyoung) [inline content/reference-ledger-format.md:244]: I know this breaks with the convention of other "source" links, but @nbougalis suggested that linking to https://github.com/ripple/rippled/blob/develop/src/ripple/app/tx/impl/Escrow.cpp might make a little more sense than that specific commit.

**Pull request merged**